app/controllers/corporate_controller.rb

```diff
@@ -18,7 +18,8 @@ def show
   end
 
   def create
-    @partner = CorporatePartner.new(partner_params)
+    @partner = CorporatePartner.find_or_initialize_by(params[:email])
```

Comment:
I feel sure I'm missing something here, but I can't google it so I'll ask — is this doing something like:

`CorporatePartner.find_or_initialize_by('email@example.com')`

What's the behaviour of `find_or_initialize_by` with a string rather than a hash?